model.py

```diff
@@ -452,36 +453,46 @@ def fetchQuote(self, instrument: Instrument) -> Quote:
         pass
 
 
+@dataclass
 class Position:
     quantityQuantization = Decimal('0.0001')
+    instrument: Instrument
+    quantity: Decimal
+    costBasis: Cash
+    averagePrice: Cash
 
     @classmethod
     def quantizeQuantity(cls, quantity: Decimal) -> Decimal:
         return quantity.quantize(cls.quantityQuantization,
                                  rounding=ROUND_HALF_EVEN)
 
-    def __init__(self, instrument: Instrument, quantity: Decimal,
-                 costBasis: Cash):
-        if instrument.currency != costBasis.currency:
+    @classmethod
+    def averagePrice(cls, instrument, quantity, costBasis) -> Cash:
+        if quantity == 0:
+            assert costBasis == 0
+            return costBasis
+
+        return costBasis / quantity / instrument.multiplier
+
+    def __post_init__(self):
+        if self.instrument.currency != self.costBasis.currency:
             raise ValueError(
                 'Cost basis {} should be in same currency as instrument {}'.
-                format(costBasis, instrument))
+                format(self.costBasis, self.instrument))
 
-        if not quantity.is_finite():
+        if not self.quantity.is_finite():
             raise ValueError(
-                'Position quantity {} is not a finite number'.format(quantity))
+                'Position quantity {} is not a finite number'.format(
+                    self.quantity))
 
-        quantity = self.quantizeQuantity(quantity)
+        quantity = self.quantizeQuantity(self.quantity)
```

Comment:
```suggestion
        self.quantity = self.quantizeQuantity(self.quantity)
```

… and omit the assignment below